src/plot_recipes.jl

```diff
@@ -411,67 +242,25 @@ julia> plot(L)
 ```
 
 To control the color of the line, use the `linecolor` keyword argument, and to
-control the color of the endpoints, use the `markercolor` keyword argument.
+control the color of the end points, use the `markercolor` keyword argument.
 To control the width, use `linewidth`.
 
 ```julia
 julia> plot(L, markercolor="green", linecolor="red", linewidth=2.)
 ```
-
-The option `add_marker` (optional, default: `true`) can be used to specify if
-endpoint markers should be plotted or not.
-```julia
-julia> plot(L, add_marker=false, linecolor="red", linewidth=2.)
-```
 """
-@recipe function plot_linesegment(L::LineSegment{N}, ε::N=zero(N);
-                                  color=:auto, alpha=0.5, add_marker=true) where {N<:Real}
-
+@recipe function plot_linesegment(L::LineSegment{N}, ε::N=zero(N)
```

Comment:
The argument `add_marker` was there to allow turning off the marker at the endpoints (The value `:none` does exist but it doesn't work..):

```julia
julia> plot(Interval(0.0, 1.0), markershape=:circle)  # ok

julia> plot(Interval(0.0, 1.0), markershape=:square)  # ok 

julia> plot(Interval(0.0, 1.0), markershape=:none)  # error
ERROR: KeyError: key :seriestype not found
Stacktrace:
 [1] getindex at ./dict.jl:478 [inlined]
 [2] preprocessArgs!(::Dict{Symbol,Any}) at /Users/forets/.julia/dev/Plots/src/args.jl:1006
 [3] #plot#140(::Base.Iterators.Pairs{Symbol,Symbol,Tuple{Symbol},NamedTuple{(:markershape,),Tuple{Symbol}}}, ::Function, ::Interval{Float64,IntervalArithmetic.Interval{Float
64}}) at /Users/forets/.julia/dev/Plots/src/plot.jl:52
 [4] (::getfield(RecipesBase, Symbol("#kw##plot")))(::NamedTuple{(:markershape,),Tuple{Symbol}}, ::typeof(plot), ::Interval{Float64,IntervalArithmetic.Interval{Float64}}) at
./none:0
 [5] top-level scope at none:0
```

Reply:
I would say this is a bug in `Plots` because we even set `seriestype := :path` in the recipe. Explicitly defining `seriestype` as in `plot(Interval(0.0, 1.0), seriestype=:path, markershape=:none)` works. Alternatively, you can use `plot(Interval(0.0, 1.0), marker=0)` (which in `GR` adds tiny dots; in `PyPlot` there are not even dots).

One reason why I removed this parameter is that one cannot (at least I did not find out how) pass keyword arguments in recursive calls to plot recipes. So we would have to introduce new recipes for each set type. And it does not work for lists of sets as well.

If you still want this parameter, I can add it back (with mild protest).

Reply:
> Explicitly defining seriestype as in plot(Interval(0.0, 1.0), seriestype=:path, markershape=:none) works

good 👍  

> If you still want this parameter, I can add it back (with mild protest).

No,we dont need it back (now or later) let's add a comment in the plot docstring explaining the previous trick.